src/services/completions.ts

```diff
@@ -1169,13 +1169,17 @@ namespace ts.Completions {
                 }
             }
 
-            // Don't suggest import completions for a commonjs-only module
-            if (preferences.includeCompletionsForModuleExports && !(sourceFile.commonJsModuleIndicator && !sourceFile.externalModuleIndicator)) {

```

Comment:
i am a bit confused now. why was not that working already? in the issue the file is not a module (i.e. no externalModuleIndicator and no commonJsModuleIndicator), so we should not have included the completions, right?

Reply:
I was thinking that even in a currently import-less file we should still add import completions if the user has `--module` set in their tsconfig. (see https://github.com/Microsoft/TypeScript/issues/22750#issuecomment-374799396)

Reply:
The reason it wasn't working already is we would always add import completions for a file with no commonjs imports/exports.

Reply:
I thought that is what https://github.com/Microsoft/TypeScript/pull/22880 was doing.. but looking at it again, the condition does not seem right.. the problem seems to be more complicated than i originally thought..

how about somethign like:

```ts
function shouldOfferImportCompletions(): boolean {
                if (preferences.includeCompletionsForModuleExports) {
                    // sure to have a module
                    if (sourceFile.externalModuleIndicator) return true;

                    // sure to have a .js file with a require, do not mess with that file
                    if (sourceFile.commonJsModuleIndicator) return false;

                    // either this file is not really a module, or it is a new file added to the project
                    // see if there are other "source" files that we can infer from
                    if (find(program.getSourceFiles(), s => !s.isDeclarationFile && !program.isSourceFileFromExternalLibrary(s) && !!s.externalModuleIndicator))
                        return true;

                    // either this is an empty project, or one with no modules in it
                    // for .js files we should fall on the safe side -- i think :)
                    if (isInJavaScriptFile(sourceFile)) return false;

                    // in the absense of any other info, then may be we can find something in the compiler options
                    if (compilerOptions.module !== undefined) return compilerOptions.module > ModuleKind.None;
                    if (compilerOptions.target >= ScriptTarget.ES2016) return true;
                    if (compilerOptions.noEmit) return true;
                }
                return false;
            }
```